### ISSUE TITLE
whois "-r" option FAILs to disable role recursion query, instead invokes RIPE db query, like BSD whois 

### ISSUE THREAD

**Issue by pgnd** [rfc1036/whois#99]
building from src (or using linux distros' rpms),
```
git clone https://github.com/rfc1036/whois.git
cd whois
git checkout next
	Already on 'next'
	Your branch is up to date with 'origin/next'.
make
ls -al ./whois
	-rwxr-xr-x 1 root root 176K Jul 14 09:55 ./whois*
./whois --version
	Version 5.5.10.
	
	Report bugs to <md+whois@linux.it>.
```

unflagged whois works
```
./whois github.com
	Domain Name: GITHUB.COM
	Registry Domain ID: 1264983250_DOMAIN_COM-VRSN
	Registrar WHOIS Server: whois.markmonitor.com
	Registrar URL: http://www.markmonitor.com
	Updated Date: 2020-09-08T09:18:27Z
	Creation Date: 2007-10-09T18:20:50Z
	Registry Expiry Date: 2022-10-09T18:20:50Z
	Registrar: MarkMonitor Inc.
	Registrar IANA ID: 292
	Registrar Abuse Contact Email: abusecomplaints@markmonitor.com
	...
```

option to NOT recurse for roles is offered
```
./whois --help | grep "\-r"
	-r                     turn off recursive look-ups for contact information
```

but instead, invokes a RIPE query
```
./whois -r github.com
	% This is the RIPE Database query service.
	% The objects are in RPSL format.
	%
	% The RIPE Database is subject to Terms and Conditions.
	% See http://www.ripe.net/db/support/db-terms-conditions.pdf

	% Note: this output has been filtered.
	%       To receive output for a database update, use the "-B" flag.

	%ERROR:101: no entries found
	%
	% No entries found in source RIPE.

	% This query was served by the RIPE Database Query Service version 1.101 (ANGUS)
```

which is the FreeBSD client's behavior,

	@ https://www.freebsd.org/cgi/man.cgi?query=whois

```
-r	     Use the R'eseaux IP Europ'eens (RIPE) database.  It contains net-
	     work numbers and domain contact information for Europe.  Get
	     query syntax documentation	using whois -r help
```

**Comment by rfc1036** [comment]
Not a bug: all command line options are supported only by number resources whois servers. Domain servers use different (and usually incompatible) syntaxes.

**Comment by pgnd** [comment]
@rfc1036 I don't understand that sentence.

Can you pls provide an example of how to query without role recursion, and NOT end up making a RIPE query?

```
./whois -r github.com
	% This is the RIPE Database query service.
	% The objects are in RPSL format.
	%
	% The RIPE Database is subject to Terms and Conditions.
	% See http://www.ripe.net/db/support/db-terms-conditions.pdf

	% Note: this output has been filtered.
	%       To receive output for a database update, use the "-B" flag.

	%ERROR:101: no entries found
	%
	% No entries found in source RIPE.

	% This query was served by the RIPE Database Query Service version 1.101 (HEREFORD)

host github.com
	github.com has address 140.82.112.3

./whois -r 140.82.112.3
	% This is the RIPE Database query service.
	% The objects are in RPSL format.
	%
	% The RIPE Database is subject to Terms and Conditions.
	% See http://www.ripe.net/db/support/db-terms-conditions.pdf

	% Note: this output has been filtered.
	%       To receive output for a database update, use the "-B" flag.

	% Information related to '140.79.0.0 - 140.83.255.255'

	% No abuse contact registered for 140.79.0.0 - 140.83.255.255

	inetnum:        140.79.0.0 - 140.83.255.255
	netname:        NON-RIPE-NCC-MANAGED-ADDRESS-BLOCK
	descr:          IPv4 address block not managed by the RIPE NCC
	remarks:        ------------------------------------------------------
	remarks:
	remarks:        For registration information,
	remarks:        you can consult the following sources:
	remarks:
	remarks:        IANA
	remarks:        http://www.iana.org/assignments/ipv4-address-space
	remarks:        http://www.iana.org/assignments/iana-ipv4-special-registry
	remarks:        http://www.iana.org/assignments/ipv4-recovered-address-space
	remarks:
	remarks:        AFRINIC (Africa)
	remarks:        http://www.afrinic.net/ whois.afrinic.net
	remarks:
	remarks:        APNIC (Asia Pacific)
	remarks:        http://www.apnic.net/ whois.apnic.net
	remarks:
	remarks:        ARIN (Northern America)
	remarks:        http://www.arin.net/ whois.arin.net
	remarks:
	remarks:        LACNIC (Latin America and the Carribean)
	remarks:        http://www.lacnic.net/ whois.lacnic.net
	remarks:
	remarks:        ------------------------------------------------------
	country:        EU # Country is really world wide
	admin-c:        IANA1-RIPE
	tech-c:         IANA1-RIPE
	status:         ALLOCATED UNSPECIFIED
	mnt-by:         RIPE-NCC-HM-MNT
	created:        2019-01-07T10:45:13Z
	last-modified:  2019-01-07T10:45:13Z
	source:         RIPE
```

**Comment by rfc1036** [comment]
This documents your options:

`
whois -h  whois.verisign-grs.com help
`

**Comment by pemensik** [comment]
Hi, I recently took maintaining whois package on Fedora. I agree current manual page is confusing and -r parameter might work different way than explained. What I understood from the source code is this flag is just passed to server along with query. It would help if special options designed to work only on whois.ripe.net would indicate it in manual. It always switches server, it should be marked somehow. Especially -r parameters looks like it reduces server usage. But it would increase usage of ripe.net servers.

**Comment by rfc1036** [comment]
I have implemented --no-recursion in 5.5.11 which should do what you want.
